heron/common/src/java/com/twitter/heron/common/utils/misc/PhysicalPlanHelper.java

```diff
@@ -301,5 +302,15 @@ public boolean isTerminatedComponent() {
   public boolean isCustomGroupingEmpty() {
     return customGrouper.isCustomGroupingEmpty();
   }
+
+  public boolean isTopologyStateful() {
+    Map<String, Object> config = topologyContext.getTopologyConfig();
+    return String.valueOf(Config.TopologyReliabilityMode.EFFECTIVELY_ONCE).equals(
```

Comment:
It is good to have a method to convert `config.get(Config.TOPOLOGY_RELIABILITY_MODE)` to enum and then compare two enum values. Comparing two Strings is error prone.

Reply:
good suggestion. Fixed.